lib/client.js

```diff
@@ -79,8 +79,14 @@ SearchSource.prototype._updateStore = function(data) {
     self.store.update(item._id, item, {upsert: true});
   });
 
-  var removedItem = _.difference(storeIds, currentIds);
-  removedItem.forEach(function(id) {
+  // Remove items in client DB that we no longer need
+  var currentIdStrs = _.map(currentIds, function(currentId){ 
+    return currentId._str; 
```

Comment:
It's possible `currentId` to be just string. So, we need to check that as well.
May be try to use the `typeof` check.